### DISCUSSION TITLE
Another useful ESO script

### DISCUSSION THREAD

**Discussion by kptb** [256shadesofgrey/eso-light-attack-weave#15] [Ideas]
Hey there,

I wonder if you'd consider another 'not cheating but useful for less dextrous' script idea I have - I think the AHK coding is a bit beyond me.

Folks have various techniques for skill bar swapping, but what about an AHK script that effectively mapped six keys (five+ultimate) to bar 1 and a different six to bar 2? Say the standard 1,2,3,4,5 and F1,F2,F3,F4,F5.

This is where AHK comes in. Basically pressing 1 would first trigger the binding to swap to bar 1, then trigger skill one, whereas F1 would trigger the binding to swap to bar 2, then trigger skill two....and so on.

Do you think this is viable in AHK? If so, is it something you'd consider pursuing, or providing some pointers on?

Thanks!

**Comment by 256shadesofgrey** [comment]
Hi,

no, I'm not interested in writing something like that myself, because I have no use for it. But if you'd like to try, these are the options I see...

If I remember correctly, there is also a "swap to X bar" keybind in ESO which is not set by default. If you set that, and use that to do the swaps, it might work. I wouldn't use the generic bar swap because it would easily desynchronize if you try to activate a skill from the other bar while stunned, while using a cast time skill (I think some ultis prevent bar swaps during their cast time) or during a bar swap cooldown. You'd still have to figure out though if it's best to always use the swap keybind when using a skill, or only sometimes, and how to handle the timings based on your approach. But if you don't implement any methods to keep the script "out of the way", it would be a very straight forward implementation (you could just start with the first version of my script, copy/paste the blocks for the additional keys, and add the corresponding bar swap as the first keypress everywhere).

Alternatively you could use [ImageSearch](https://www.autohotkey.com/docs/commands/ImageSearch.htm) to track which bar you're on. Because AHK already has such functionality, it could make the logic for tracking the bar simpler (you could even use the generic bar swap in this case), but it would make your script highly UI dependent. Things like addons, UI scale and even screen resolution could all mess with this approach, so you would either have to read out a lot of parameters from your system when starting the script, or have a long list of options to adjust before you could use this script on someone else's client.

**Reply by kptb** [reply]
Thanks for the quick and helpful response!
